### PR TITLE
Add the 1.2.7 Release Note to Dev Site

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the 1.2.7 Release Note to Dev Site.
> Fixes #

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 1]
```diff
@@ -22,9 +22,92 @@ However, you need to use the following commands instead of the above if you have
...
+- The mocking API also supports scoping and stubbing of mock functions that are declared for functions in imported modules.
+
+Function mocking is done by using the following functions.
+
+- The `test:when(mockObj)` function is used to initialize the mocking capability within a particular test case.
+- This allows you to use the associated mocking functions like `call()`, `thenReturn()`, and `withArguments()`.
+
+### Improvements related to code actions
+
+#### Introducing an AI-based `Data Mapping` Code Action
```
Comment (malithjayasinghe) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:61]: This should be release with SwanLake not 1.2.7 - we need to remove this from here

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 2]
```diff
@@ -22,9 +22,92 @@ However, you need to use the following commands instead of the above if you have
 
 If you have not installed jBallerina, then download the [installers](https://ballerina.io/downloads/) to install.
 
+## Language
```
Comment (anupama-pathirage) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:25]: If we don't have Language side fixes shall we remove this topic?

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 3]
```diff
@@ -22,9 +22,92 @@ However, you need to use the following commands instead of the above if you have
...
+
+When a CSV file has an empty record, the previous implementation throws a null pointer exception and it was fixed with this release.
+
+## Developer Tools
+
+### Test Framework
+
+#### Object Mocking
+
+Object mocking enables controlling the values of member variables and the behavior of the member functions of an object. The following improvmenets are introduced with regard to this.
```
Comment (anupama-pathirage) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:39]:
```suggestion
Object mocking enables controlling the values of member variables and the behavior of the member functions of an object. The following improvements are introduced with regard to this.
```

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 4]
```diff
@@ -22,9 +22,92 @@ However, you need to use the following commands instead of the above if you have
...
+## Language
+
 ## Standard Library
-- 
 
-## Dev Tools
-- 
+### I/O
+
+When a CSV file has an empty record, the previous implementation throws a null pointer exception and it was fixed with this release.
```
Comment (anupama-pathirage) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:31]: NPE is java level issue. Should we re-word it with something related to ballerina error?

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md (hunk 1 above)
Comment (praneesha) [reply downloads/release-notes/1.2.7/RELEASE_NOTE.md]: Done.

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 5]
```diff
@@ -23,8 +23,33 @@ However, you need to use the following commands instead of the above if you have
 If you have not installed jBallerina, then download the [installers](https://ballerina.io/downloads/) to install.
 
 ## Standard Library
-- 
 
-## Dev Tools
-- 
+### I/O
 
+When a CSV file has an empty record, the previous implementation throws a null pointer exception and it was fixed with this release.
```
Comment (ThisaruGuruge) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:29]:
```suggestion
When a CSV file has an empty record, the previous implementation panicked. This is fixed with this release.
```

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md (hunk 2 above)
Comment (praneesha) [reply downloads/release-notes/1.2.7/RELEASE_NOTE.md]: Done.

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 6]
```diff
@@ -23,8 +23,33 @@ However, you need to use the following commands instead of the above if you have
 If you have not installed jBallerina, then download the [installers](https://ballerina.io/downloads/) to install.
 
 ## Standard Library
-- 
 
-## Dev Tools
-- 
+### I/O
 
+When a CSV file has an empty record, the previous implementation panicked. This is fixed with this release.
```
Comment (ThisaruGuruge) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:29]:
```suggestion
- When a CSV file has an empty record, the previous implementation panicked. This is fixed with this release.
```

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md (hunk 4 above)
Comment (praneesha) [reply downloads/release-notes/1.2.7/RELEASE_NOTE.md]: Done via https://github.com/ballerina-platform/ballerina-dev-website/pull/1140#discussion_r469204240.

File: downloads/release-notes/1.2.7/RELEASE_NOTE.md [hunk 7]
```diff
@@ -2,7 +2,7 @@
 layout: release-note
 title: Release note
 ---
-# Overview of jBallerina {{ site.data.stable-latest.metadata.version }}
+## Overview of jBallerina {{ site.data.stable-latest.metadata.version }}
```
Comment (ThisaruGuruge) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:5]: The version is not replaced. Please check for the other places as well. `{{ site.data.stable-latest.metadata.version }}` should be replaces with `1.2.7`

Comment (ThisaruGuruge) [inline downloads/release-notes/1.2.7/RELEASE_NOTE.md:5]: The version is not replaced. Please check for the other places as well. `{{ site.data.stable-latest.metadata.version }}` should be replaces with `1.2.7`

Comment (praneesha) [reply downloads/release-notes/1.2.7/RELEASE_NOTE.md:5]: @shafreenAnfar - Do we need to change these manually?

Comment (ThisaruGuruge) [reply downloads/release-notes/1.2.7/RELEASE_NOTE.md:5]: Marking as resolved as discussed.

**Pull request merged**